src/getDailySummary.js

```diff
@@ -13,7 +13,7 @@ const getDailySummary = async (date = new Date()) => {
   }
 
   // PAGING DR.HACKY AS FUCK, PAGING DR.HACKY AS FUCK
-  const formattedDate = date.toISOString().split('T')[0];
+  const formattedDate = date.toLocaleDateString();
```

Comment:
I apologize about the confusion from my previous comment - the `formattedDate` variable is used to call the Wakatime API via the `getMySummary` method on line 19.

I think what you want to modify is [`line 12` of the `generateDailySummary`](https://github.com/jaebradley/wakatime-cli/blob/master/src/services/generateDailySummary.js#L12) method and [`line 18` in `generateWeeklySummary`](https://github.com/jaebradley/wakatime-cli/blob/master/src/services/generateWeeklySummary.js#L18).

Reply:
It looks like the result date(time)s get returned as UTC times in the user's timezone setting (my TZ settings is `America/New_York` but I'm in Mexico right now)?

![image](https://user-images.githubusercontent.com/8136030/43841846-af0bf04a-9ae9-11e8-8876-daad88e08201.png)

![image](https://user-images.githubusercontent.com/8136030/43841908-cd3c51cc-9ae9-11e8-8b03-099535bfc661.png)

Does that mean that formatting needs to be done after receiving the response?

* [Wakatime API](https://wakatime.com/developers#summaries)

Reply:
I think the time difference has been taken care of on <b>Wakatime API</b> side. The excerpt below is the API response I get for yesterday's summary (it clearly reflects my time zone by returning a UTC string that is pushed up by 7 hours), and I suppose the <b>Wakatime-client</b> you wrote should be getting the same API response. (ps. Good job on writing that client)

<img width="1351" alt="screen shot 2018-08-08 at 10 41 37 am" src="https://user-images.githubusercontent.com/9512759/43854589-2c52d484-9af8-11e8-8c7c-576bed578413.png">

So even though Wakatime stores users' time-related information in UTC format, it always returns the API response appropriate to users' local time according to the time zone stored on their settings (see below). That's why users in Vancouver gets the response that's pushed up by 7 hours.

<img width="200" alt="screen shot 2018-08-08 at 10 53 18 am" src="https://user-images.githubusercontent.com/9512759/43855307-07b9beec-9afa-11e8-89f7-ee14b09e7fb0.png">

Now the problem is with the request we sent to <b>Wakatime API</b>. If we use `toISOString()`, it is sending UTC, but Wakatime doesn't know that. It's treating it as in Vancouver time, so as usual it returns the response that's 7 hours pushed-up, which becomes 7 hours ahead of what users in Vancouver requests. This starts to become a problem when Vancouver users are close to the end of the day, when the request sent at 9pm (2018/08/06) returns the date that is 4am in the next day (2018/08/07). This has effects on your `waka today`, `waka yesterday` and `waka week` commands, because they are getting some or all of their summaries from the future.

As you can see, this problem would only start to reveal the further users are aways from Greenwich time zone, because there's a bigger proportion of day in which they get their summaries from the future or the past. You could also argue that most users wouldn't bother with specifying time zone in the settings, but I think that settings is automatically enforced and collected from the user's computer by Wakatime. To me it's just a relief to know there's one less thing to worry about.

Lucky for us it should now be an easy fix with using `toLocaleDateString`, while letting Wakatime handles time differences.

Reply:
@hankchanocd thanks for that really detailed response, but my question was getting at what happens if the user's locale does not share the timezone they specified in their settings? (i.e. I set my user timezone to Vancouver (`-7`), but then travel to Boston (`-3`)?)

Will using `toLocaleDateString` still produce out-of-date data because at 1 AM in Boston it'll still be the previous day in Vancouver, so executing `waka today` in Boston will return no results because based on the user's specified time zone, no activities have been recorded for the "next" day?

Reply:
As I played around with the settings earlier today, I noticed Wakatime would only give me a warning on the homepage when there's a divergence between the time zone I specified and the time zone it detected from my computer, but it's left to me to manually change the time zone. So in the less often scenario of traveling to different time zones, user would have to manually update their settings on the website. In that case, you can argue there's a need to write another module to automate the process of time conversion between time zones, specifically for the those who travel a lot. But I think my job here is done. I'm just happy that `toLocaleTimeString()` works perfectly on my computer.

As for the Boston-Vancouver scenario in the second paragraph, I assume you are referring to teamwork in different time zones. Because Wakatime stores everything in UTC format, I think it simply converts the response to the user's time zone. But don't quote me on that, I have no experience with Wakatime team features. 

It's been fun chatting with you. Keep up the good work!